### ISSUE TITLE
Training procedure

### ISSUE THREAD

**Issue by edwardnguyen1705** [LiheYoung/Depth-Anything#33]
Dear @LiheYoung ,

First, thank you for sharing such a great work. The feature alignment technique to preserve semantic info is very interesting. I am very interested in the training techniques, so I would like to ask you some questions.

From the sub-section "4.1. Implementation Details", as I understand the training procedure goes like this:
1. train a teacher model (pretrained weights from Segformer?) T for 20 epochs on labeled images, only horizontal flipping aug is used.
2. jointly train labeled and unlabeled images
    2.0. Augmentation techniques:  horizontal flipping, two forms of perturbations (strong color distortions and CutMix)
    2.1. re-initialize S (that means pretrained weights are from Segformer?)
    2.2. "train a student model to sweep across all unlabeled images for one time": train just one epoch?
    2.3. continue to train the student model for how many epochs?
    
Could you confirm my understanding is correct? Thanks for your time.

**Comment by juseonghan** [comment]
I had a follow up question about this - the loss function specified in the paper for semantic preservation (equation 9) computes the sum of all the cosine similarity terms over each pixel - but isn't one feature vector produced for one image? How do you produce one cosine similarity term for each pixel that you can mean over them? Hope you can clarify, thank you!

**Comment by tcourat** [comment]
> I had a follow up question about this - the loss function specified in the paper for semantic preservation (equation 9) computes the sum of all the cosine similarity terms over each pixel - but isn't one feature vector produced for one image? How do you produce one cosine similarity term for each pixel that you can mean over them? Hope you can clarify, thank you!

Hi.
DinoV2 encoder output a 37x37 feature maps=1369 feature vectors (base image have size 518x518, and patch size is 14x14) so I guess the feature alignment loss is computed at the feature map level, not the pixel level.

**Comment by sijieaaa** [comment]
> Dear @LiheYoung ,
> 
> First, thank you for sharing such a great work. The feature alignment technique to preserve semantic info is very interesting. I am very interested in the training techniques, so I would like to ask you some questions.
> 
> From the sub-section "4.1. Implementation Details", as I understand the training procedure goes like this:
> 
> 1. train a teacher model (pretrained weights from Segformer?) T for 20 epochs on labeled images, only horizontal flipping aug is used.
> 2. jointly train labeled and unlabeled images
>    2.0. Augmentation techniques:  horizontal flipping, two forms of perturbations (strong color distortions and CutMix)
>    2.1. re-initialize S (that means pretrained weights are from Segformer?)
>    2.2. "train a student model to sweep across all unlabeled images for one time": train just one epoch?
>    2.3. continue to train the student model for how many epochs?
> 
> Could you confirm my understanding is correct? Thanks for your time.

Me confusion too.

Why not directly use "strong perturbations" in the teacher training stage?
